DataFormats/Detectors/FIT/FDD/include/DataFormatsFDD/RecPoint.h

```diff
@@ -15,23 +15,77 @@
 #define ALICEO2_FDD_RECPOINT_H
 
 #include "CommonDataFormat/InteractionRecord.h"
+#include "CommonDataFormat/InteractionRecord.h"
+#include "CommonDataFormat/TimeStamp.h"
+#include "DataFormatsFDD/ChannelData.h"
+#include "CommonDataFormat/RangeReference.h"
+#include "DataFormatsFDD/Digit.h"
 
 namespace o2
 {
 namespace fdd
 {
+struct ChannelDataFloat {
+
+  int mPMNumber = -1;         // channel Id
+  double mTime = -20000;      // time in ps, 0 at the LHC clk center
+  double mChargeADC = -20000; // charge [channels]
+  int adcId = -1;             // QTC chain
 
-struct RecPoint {
+  ChannelDataFloat() = default;
+  ChannelDataFloat(int Channel, double Time, double Charge, int AdcId)
+  {
+    mPMNumber = Channel;
+    mTime = Time;
+    mChargeADC = Charge;
+    adcId = AdcId;
+  }
 
-  double mMeanTimeFDA = o2::InteractionRecord::DummyTime;
-  double mMeanTimeFDC = o2::InteractionRecord::DummyTime;
+  void print() const;
+
+  ClassDefNV(ChannelDataFloat, 1);
+};
 
-  o2::InteractionRecord mIntRecord; // Interaction record (orbit, bc) from digits
+class RecPoint
+{
 
+ public:
+  enum TimeTypeIndex : int { TimeA,
+                             TimeC };
   RecPoint() = default;
-  RecPoint(double timeA, double timeC, o2::InteractionRecord iRec) : mMeanTimeFDA(timeA), mMeanTimeFDC(timeC), mIntRecord(iRec) {}
+  RecPoint(const std::array<int, 2>& collisiontime, int first, int ne,
+           o2::InteractionRecord iRec, o2::fdd::Triggers triggers)
+    : mCollisionTimePs(collisiontime)
+  {
+    mRef.setFirstEntry(first);
+    mRef.setEntries(ne);
+    mIntRecord = iRec;
+    mTriggers = triggers;
+  }
+  ~RecPoint() = default;
+
+  float getCollisionTime(TimeTypeIndex type) const { return mCollisionTimePs[type]; }
+  float getCollisionTimeA() const { return getCollisionTime(TimeA); }
+  float getCollisionTimeC() const { return getCollisionTime(TimeC); }
+  bool isValidTime(TimeTypeIndex type) const { return getCollisionTime(type) < sDummyCollissionTime; }
+  void setCollisionTime(Float_t time, TimeTypeIndex type) { mCollisionTimePs[type] = time; }
+
+  o2::fdd::Triggers getTrigger() const { return mTriggers; }
```

Comment:
Return const ref instead of a copy: `const o2::fdd::Triggers& getTrigger() const { return mTriggers; }`

Reply:
Done